### DISCUSSION TITLE
xrdp devel branch gives X server could not be started /  Fails to take device /dev/dri/card* error on Debian 12

### DISCUSSION THREAD

**Discussion by eugrus** [neutrinolabs/xrdp#3131] [General]
I've compiled and installed the H.264-abled (devel) branch of xrdp.

It gives out the "X server could not be started error" on login.

In ``.xorgxrdp.10.log`` I've noticed the following:
```
[  9415.107] (II) systemd-logind: took control of session /org/freedesktop/login1/session/_320
[  9415.109] (II) xfree86: Adding drm device (/dev/dri/card1)
[  9415.109] (II) Platform probe for /sys/devices/pci0000:00/0000:00:01.1/0000:02:00.0/drm/card1
[  9415.109] (EE) systemd-logind: failed to take device /dev/dri/card1: Operation not permitted
[  9415.109] (II) xfree86: Adding drm device (/dev/dri/card0)
[  9415.109] (II) Platform probe for /sys/devices/pci0000:00/0000:00:02.0/drm/card0
[  9415.109] (EE) systemd-logind: failed to take device /dev/dri/card0: Operation not permitted
[  9415.119] (--) PCI:*(0@0:2:0) 8086:0416:17aa:221d rev 6, Mem @ 0xf1400000/4194304, 0xe0000000/268435456, I/O @ 0x00004000/64, BIOS @ 0x????????/131072
[  9415.119] (--) PCI: (2@0:0:0) 10de:1290:17aa:221d rev 161, Mem @ 0xf0000000/16777216, 0xc0000000/268435456, 0xd0000000/33554432, I/O @ 0x00003000/128, BIOS @ 0x????????/524288
[  9415.119] (II) "glx" will be loaded. This was enabled by default and also specified in the config file.
```

Is that where the problem lies and how do I solve it?

My GPUs are Intel Corporation 4th Gen Core Processor Integrated Graphics Controller and NVIDIA Corporation GK208M [GeForce GT 730M].

**Comment by akarl10** [comment]
Probably Xorg wants to add every gpu. This is not possible in a headless session. Also xorgxrdp and xrdp must match
* if not done already also compile xorgxrdp (https://github.com/neutrinolabs/xorgxrdp)
* replace /etc/X11/xrdp/xorg.conf with https://github.com/neutrinolabs/xorgxrdp/blob/devel/xrdpdev/xorg.conf (make a backup if you made some changes to that file)
* Since you are on debian: check if `param=/usr/lib/xorg/Xorg` is the path you are using under the `[Xorg]` section in `/etc/xrdp/sesman.ini`

**Reply by eugrus** [reply]
Thank you! This did help out!
